templates/common/render/render.go

```diff
@@ -389,19 +389,18 @@ type stepParams struct {
 	// ignored while being copied to destination directory.
 	ignorePatterns []model.String
 
-	// includedFromDest is a list of every file (no directories) that was copied
-	// from the destination directory into the scratch directory. We want to
-	// track these because they are treated specially in the final phase of
-	// rendering. When we commit the template output from the scratch directory
-	// into the destination directory, these paths are always allowed to be
-	// overwritten. For other files not in this list, it's an error to try to
-	// write to an existing file. This whole scheme supports the feature of
-	// modifying files that already exist in the destination.
-	//
-	// These are paths relative to the --dest directory (which is the same thing
-	// as being relative to the scratch directory, the paths within these dirs
-	// are the same).
-	includedFromDest map[string]struct{}
+	// includedFromDest tracks files (no directories) that were copied from the
+	// destination directory into the scratch directory. The map keys are the
+	// location of the file in the scratch directory, and the map values are the
+	// directory from which the file was taken (sometimes this gets complicated
+	// there are multiple "destination" directories. We want to track these
+	// because they are treated specially in the final phase of rendering. When
+	// we commit the template output from the scratch directory into the
+	// destination directory, these paths are always allowed to be overwritten.
+	// For other files not in this list, it's an error to try to write to an
+	// existing file. This whole scheme supports the feature of modifying files
+	// that already exist in the destination.
+	includedFromDest map[string]string
```

Comment:
Just a question.

I haven't used this feature before. Say I have existing files in the dest folder that I'd like to keep, what will happen if I don't specify them via `includedFromDest`? Put it differently, when would I want to `includedFromDest`

Reply:
As a template developer, the reason you'd want to use "include from destination" is because you want the template to modify a file in place. The canonical use case is `outputs.tf`: in the spec file, you'd "include-from-destination" the preexisting outputs.tf file, do a string replacement or append operation to add a new terraform output, and then the modified file would overwrite the old file.